core/domain/app_feedback_report_domain.py

```diff
@@ -274,8 +312,16 @@ def get_android_report_from_dict(
                 user_supplied_feedback_json['user_feedback_selected_items'],
                 user_supplied_feedback_json['user_feedback_other_text_input']))
 
-        system_context_json = report_dict['system_context']
-        device_context_json = report_dict['device_context']
+        # Here we use MyPy ignore because 'system_context' key is not defined
+        # in AndroidFeedbackReportDict and to_dict() but still we are accessing
+        # this key which causes MyPy to throw a error. Thus to avoid the error,
+        # we used ignore here.
+        system_context_json = report_dict['system_context']  # type: ignore[misc]
+        # Here we use MyPy ignore because 'device_context' key is not defined
+        # in AndroidFeedbackReportDict and to_dict() but still we are accessing
+        # this key which causes MyPy to throw a error. Thus to avoid the error,
+        # we used ignore here.
+        device_context_json = report_dict['device_context']  # type: ignore[misc]
```

Comment:
@seanlip @aasiffaizal can any of you please help me with this?

Basically, the situation here is that we are expecting `device_context`, `platform_type`, `report_submission_timestamp_sec`, and many other keys but these keys are not returned by any of the to_dict methods. So, My query is that from where these values are coming from I tried to search for these keys but the max I found is that these values are used in tests but there is no method that returns these keys in the dictionary. 

Taking an example of `platform_type` key:

AppFeedbackReport class's to_dict() method return a dict with `platform` key
https://github.com/oppia/oppia/blob/d4b06399ad2859df6917c5e18ce9024e452351d0/core/domain/app_feedback_report_domain.py#L106
But on other hand AppFeedbackReport class's from_dict method expecting a `platform_type` key.
https://github.com/oppia/oppia/blob/d4b06399ad2859df6917c5e18ce9024e452351d0/core/domain/app_feedback_report_domain.py#L231

even if we look at the docstring i think they are pointing to same dict -- please correct me If I'm missing something!!

Reply:
I think this hasn't been joined to the frontend yet, which is why there's a bit of difficulty here. This is what is supposed to happen:

- Feedback dicts may be submitted through either web or Android.
- The fields in each dict will depend on what the platform_type of the dict is. Web feedback dicts have certain fields and Android feedback dicts have certain fields.

I also think that from_dict() in app_feedback_report_domain.py is not well-named. It is not the usual from_dict() method which converts a dict of the current object into the corresponding domain object -- instead, what this method is doing is taking a "report dict" from outside and constructing the general domain object for it. So perhaps the best thing to do there is rename from_dict() to from_submitted_feedback_dict() or similar, to clarify that these are different concepts.

Does this help?

Reply:
Yeah, I did! and changed the function name!

Reply:
The function's name is reverted from from_submitted_feedback_dict to from_dict because it is causing backend tests to fail.